src/main/java/org/elasticsearch/action/admin/indices/mapping/delete/TransportDeleteMappingAction.java

```diff
@@ -109,7 +109,14 @@ protected void doExecute(DeleteMappingRequest request, ActionListener<DeleteMapp
 
     @Override
     protected ClusterBlockException checkBlock(DeleteMappingRequest request, ClusterState state) {
-        return state.blocks().indicesBlockedException(ClusterBlockLevel.METADATA, state.metaData().concreteIndices(request.indicesOptions(), request.indices()));
+        String[] concreteIndices = state.metaData().concreteIndices(request.indicesOptions(), request.indices());
+        ClusterBlockException status = state.blocks().indicesBlockedException(ClusterBlockLevel.METADATA_WRITE, concreteIndices);
+        if (status != null) {
+            return status;
+        }
+        // DeleteMapping executes a DeleteByQuery as soon as the mapping is deleted,
+        // so we need to check the same block as DeleteByQuery here
+        return state.blocks().indicesBlockedException(ClusterBlockLevel.WRITE, concreteIndices);
```

Comment:
Yes, I will add it again once I rebase (again) this PR.

> Also, how did you find this issue?

The test fails, then I dig into the code and see that DeleteMapping executes a DeleteByQuery under the hood.